autowebcompat/utils.py

```diff
@@ -140,32 +140,40 @@ def next(self):
 
 
 def balance(it):
-    last_label = None
-    queue_1 = []
-    queue_0 = []
+    # Initialise last_label to not (label of first element)
+    last_label = it[0][1]
```

Comment:
The function needs to support an iterator, `it` could be a list or an iterator.
You should add a test case that passes to balance an iterator, so you can make sure it's working in both cases.